### DISCUSSION TITLE
Serverless Functions timeout when switching to Node 16.x or 18.x

### DISCUSSION THREAD

**Discussion by defsteph** [vercel/community#3179] [Help] [answered]
### Summary

I've been running my site on Node 14.x for quite a while, saw a warning that it was EOL, and thought I'd upgrade. I've been running 19 locally on my machine, so switched to 18 locally, and everything worked just fine, using both `yarn dev` as well as `yarn build` followed by `yarn start`.

When I deploy my site to Vercel, all my Serverless Functions start to timeout. Switching back to Node 14.x makes them all magically start to work again. Same issue arises when switching to Node 16.x. Then only Node version I can get my Serverless Functions to run on in Vercel, is 14.x

I can't find any trace in my code of anything relating to the node version.

I am using Cloudflare to proxy my site, because reasons, but I've double and triple checked the settings, and made no changes because of the Node version change.

_Update:_ I'm using `fetch` in my API routes, to proxy some APIs. However, I'm not using `node-fetch` or any other wrapper around `fetch`, so these should reasonably fail in Node 14.x (but work fine apparently), and work fine in Node 18.x (but could very well be what's breaking)

### Steps to Reproduce

I have no idea on how to reproduce this. Judging by experience, it's probably some super obvious thing, once you know what it is...

**Comment by amyegan** [comment, marked as answer]
Sorry to hear you're having trouble with new Node versions. There may be more information available in the [Runtime Logs](https://vercel.com/docs/concepts/observability/runtime-logs#). 

There are many possible causes of errors, so I'll need to know more about your application to help narrow them down. Do you have a public repo that you can share here?

**Reply by defsteph** [reply]
Unfortunately not. But I've started upgrading to Next 13, and with _almost_ the exact same code, things are looking up!

**Comment by ydax** [comment]
@amyegan Has Vercel published a blog post or tutorial on managing this upgrade process? If so, could you please share a link to it?

**Reply by amyegan** [reply]
Hey @ydax. If you're looking for a Next.js upgrade guide, you can find that at https://nextjs.org/docs/pages/building-your-application/upgrading

If you're asking about upgrading the Node.js version for your project, you can do that in project settings. You can find out more about that at https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js#node.js-version

**Comment by ydax** [comment]
Okay, thanks. I mean, is this as easy as changing a setting in Vercel or do we need to refactor in our codebase?

**Reply by amyegan** [reply]
It's usually as easy as changing a setting. The new version will be used for the next deployment. 

Occasionally, some package versions will not be compatible with a newer version of Node and will also need to be updated. If that is the case, you'll see a message about it in the build log to let you know what else needs to be updated.

**Comment by erwinsmit** [comment]
Hi,

I'm having the same problem. After updating to Node version 16 or 18 all my API routes timeout. Locally everything works fine on Node version 16 or 18.

The logs are not giving me much information. Except that there is a 504:
![image](https://github.com/vercel/community/assets/5425589/571aa44e-43b0-4a65-b2e9-6492d535f4d3)

I tried disabling all my api routes and create a simple hello world api route, that also hangs.
```
export default function handler (req, res) {
    res.status(200).json({message: 'hello world'})
}
```

The weird thing is, I copied the same code to another Vercel project, and it works fine there.

https://www.tuinstadstaalwijk.nl/api/test/simple (timeout)
https://erwin-sandbox.vercel.app/api/test (works fine, same code)

**Reply by erwinsmit** [reply]
After some more digging in the github discussion history, I found it was MUI icons that caused the problem https://github.com/orgs/vercel/discussions/2447
